### PR TITLE
Format --use-git flag

### PR DESCRIPTION
**Pull request by esharf**
## Status
In progress.

## Related Issues
[#21103](https://github.com/demisto/etc/issues/21103)

## Description
adding a --use-git flag for the format command.

### CODE REVIEW

File: demisto_sdk/main.py [hunk 1]
```diff
@@ -288,21 +291,37 @@ def lint(config, dir, **kwargs):
...
+        for _file in files:
+            _file = _file['name']
+            file_type = 'integration' if checked_type(_file, YML_ALL_INTEGRATION_REGEXES)\
+                else 'script' if checked_type(_file, YML_ALL_SCRIPTS_REGEXES)\
+                else 'playbook' if checked_type(_file, YML_ALL_PLAYBOOKS_REGEX)\
+                else None
+            if file_type:
+                res = file_type_and_linked_class[file_type](source_file=_file).format_file()
+                if res:
+                    return res
```
Comment (ronykoz) [inline demisto_sdk/main.py]: add this logic to the class please

File: demisto_sdk/git_tools.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from demisto_sdk.common.tools import run_command
+from typing import Callable, List
+import re
```
Comment (ronykoz) [inline demisto_sdk/git_tools.py]: order them by pep convention
import re
from typing import Callable, List

from demisto_sdk.common.tools import run_command

File: demisto_sdk/git_tools.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Git:
```
Comment (ronykoz) [inline demisto_sdk/git_tools.py]: not sure that this should be a class

File: demisto_sdk/main.py [hunk 4]
```diff
@@ -15,15 +16,12 @@
...
 from demisto_sdk.yaml_tools.content_creator import ContentCreator
-from demisto_sdk.yaml_tools.update_playbook import PlaybookYMLFormat
 from demisto_sdk.json_to_outputs.json_to_outputs import json_to_outputs
-from demisto_sdk.yaml_tools.update_integration import IntegrationYMLFormat
 from demisto_sdk.common.constants import SCRIPT_PREFIX, INTEGRATION_PREFIX
 
-from demisto_sdk.test_playbook_generator.test_playbook_generator import TestPlaybookGenerator
 
+from demisto_sdk.test_playbook_generator.test_playbook_generator import TestPlaybookGenerator
 pass_config = click.make_pass_decorator(DemistoSDK, ensure=True)
```
Comment (ronykoz) [inline demisto_sdk/main.py]: separate the imports from the code itself

File: demisto_sdk/yaml_tools/format_module.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from demisto_sdk.common.constants import YML_ALL_INTEGRATION_REGEXES, YML_ALL_SCRIPTS_REGEXES, YML_ALL_PLAYBOOKS_REGEX
+from demisto_sdk.git_tools import Git
+from demisto_sdk.yaml_tools.update_playbook import PlaybookYMLFormat
+from demisto_sdk.yaml_tools.update_script import ScriptYMLFormat
+from demisto_sdk.yaml_tools.update_integration import IntegrationYMLFormat
+from demisto_sdk.common.scripts.update_id_set import checked_type
+from demisto_sdk.common.tools import print_error
+
+
+def format_command(kwargs):
```
Comment (ronykoz) [inline demisto_sdk/yaml_tools/format_module.py]: lets call it format_manager

Comment (ronykoz) [reply demisto_sdk/yaml_tools/format_module.py]: shouldn't it be **kwargs? and add docstring

File: demisto_sdk/git_tools.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    return branch_name_reg.group(1)
+
+
+def get_changed_files(from_branch: str = 'master', filter_results: Callable = None):
+    temp_files = run_command(f'git diff --name-status {from_branch}').split('\n')
+    files: List = []
+    for file in temp_files:
+        if file:
+            files.append({
+                'name': file[2:],
```
Comment (ronykoz) [inline demisto_sdk/git_tools.py]: if you have renamed files this will cause problems

Comment (esharf) [reply demisto_sdk/git_tools.py]: fixed

**Comment by ronykoz** [comment]
Update the CHANGELOG file too please

**Pull request merged**